docs/library/machine.DAC.rst

```diff
@@ -0,0 +1,66 @@
+.. currentmodule:: machine
+.. _machine.DAC:
+
+class DAC -- digital to analog conversion
+=========================================
+
+The DAC is used to output an analog voltage based on a digital value.
+
+The output voltage will be between 0 and 3.3V.
+
+DAC is currently supported on ESP32 [#esp32_dac]_, SAMD, Renesas RA and STM32
+(via :ref:`pyb.DAC <pyb.DAC>`).
+
+Example usage (ESP32)::
+
+    from machine import DAC
+
+    dac = DAC(pin)    # create a DAC object acting on a pin
+    dac.write(128)    # write a value to the DAC
+    dac.write(255)    # output maximum value, 3.3V
+
+Constructors
+------------
+
+.. class:: DAC(id)
+
+  Construct a new DAC object.
+
+  ``id`` is a pin object (ESP32 and Renesas RA) or an index to a DAC resource
```

Comment:
1. There is no machine.DAC for STM32. It's pyb.DAC, and it has more features like write_timed.
2. On ESP32 and Renesas and STM32 `id` must be a value that refers to a pin. It may be a Pin object, but can as well be a number or string (Renesas, STM32), as supported by the port.

Reply:
Updated to remove any mention of stm32 except for a note that refers to 'similar functionality' in `pyb,DAC`.

Reply:
Using "pin object" is consistent with most of the other parts of the docs, including `machine.ADC`.  So I think it's fine to use here as well.